pkg/cache/sql/integration_test.go

```diff
@@ -0,0 +1,324 @@
+		},
+		{
+			name: "not eq filter",
+			filters: orFiltersForFilters(informer.Filter{
+				Field:   []string{`metadata`, `annotations[somekey]`},
+				Match:   "somevalue",
+				Op:      informer.NotEq,
+				Partial: false,
+			}),
+			wantNames: []string{"partial-matches", "not-matches-filter", "missing"},
+		},
+		{
+			name: "partial not eq filter",
+			filters: orFiltersForFilters(informer.Filter{
+				Field:   []string{`metadata`, `annotations[somekey]`},
+				Match:   "somevalue",
+				Op:      informer.NotEq,
+				Partial: true,
+			}),
+			wantNames: []string{"not-matches-filter", "missing"},
+		},
+		{
+			name: "multiple or filters match",
+			filters: orFiltersForFilters(
+				informer.Filter{
+					Field:   []string{`metadata`, `annotations[somekey]`},
+					Match:   "somevalue",
+					Op:      informer.Eq,
+					Partial: true,
+				},
+				informer.Filter{
+					Field:   []string{`metadata`, `annotations[somekey]`},
+					Match:   "notequal",
+					Op:      informer.Eq,
+					Partial: false,
+				},
+			),
+			wantNames: []string{"matches-filter", "partial-matches", "not-matches-filter"},
+		},
+		{
+			name: "or filters on different fields",
+			filters: orFiltersForFilters(
+				informer.Filter{
+					Field:   []string{`metadata`, `annotations[somekey]`},
+					Match:   "somevalue",
+					Op:      informer.Eq,
+					Partial: true,
+				},
+				informer.Filter{
+					Field:   []string{`metadata`, `name`},
+					Match:   "missing",
+					Op:      informer.Eq,
+					Partial: false,
+				},
+			),
+			wantNames: []string{"matches-filter", "partial-matches", "missing"},
+		},
+		{
+			name: "and filters, both must match",
+			filters: []informer.OrFilter{
+				{
+					Filters: []informer.Filter{
+						{
+							Field:   []string{`metadata`, `annotations[somekey]`},
+							Match:   "somevalue",
+							Op:      informer.Eq,
+							Partial: true,
+						},
+					},
+				},
+				{
+					Filters: []informer.Filter{
+						{
+							Field:   []string{`metadata`, `name`},
+							Match:   "matches-filter",
+							Op:      informer.Eq,
+							Partial: false,
+						},
+					},
+				},
+			},
+			wantNames: []string{"matches-filter"},
+		},
+		{
+			name: "no matches",
+			filters: orFiltersForFilters(
+				informer.Filter{
+					Field:   []string{`metadata`, `annotations[somekey]`},
+					Match:   "valueNotRepresented",
+					Op:      informer.Eq,
+					Partial: false,
+				},
+			),
+			wantNames: []string{},
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		i.Run(test.name, func() {
+			options := informer.ListOptions{
+				Filters: test.filters,
+			}
+			partitions := []partition.Partition{defaultPartition}
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+			cfgMaps, total, continueToken, err := cache.ListByOptions(ctx, options, partitions, testNamespace)
+			i.Require().NoError(err)
+			// cancel here so that we don't build up a large list of defer'd cancels
+			cancel()
```

Comment:
nit: I think deferring the cancel would be ok here, because it would fire when leaving the anonymous function?